sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3503,6 +3504,25 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     checkAnswer(sql("select CAST(-32768 as short) DIV CAST (-1 as short)"),
       Seq(Row(Short.MinValue.toLong * -1)))
   }
+
+
+  test("SPARK-31916: verify that PlanStringConcat's output shows the actual length of the plan") {
```

Comment:
Since `PlanStringConcat's` is in the catalyst package, could you move this tests there?